crates/driver/src/domain/mempools.rs

```diff
@@ -235,14 +240,26 @@ pub enum RevertProtection {
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error("Mined reverted transaction: {0:?}")]
-    Revert(eth::TxId),
+    #[error("Mined reverted transaction: {tx_id:?}")]
+    Revert {
+        tx_id: eth::TxId,
+        block_no: Option<BlockNo>,
```

Comment:
nit: I'm not stoked on the naming. Either simply `block` or `block_number` seems more appropriate.